Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/ArgumentSpace.java

```diff
@@ -0,0 +1,352 @@
+     * property bindings.
+     */
+    public void snapToKnotIndex() {
+        int kti = getKnotIndex();
+        if (kti > 0 && kti <= leftArguments.size()) {
+            Region arg = leftArguments.get(kti-1); // First argument left of the knot.
+
+            // Binds the knot to the first argument on the left, so that if
+            // there is a length change to the left of the knot (because of a
+            // Label's text change), the knot moves accordingly.
+            knot.layoutXProperty().bind(arg.layoutXProperty().add(arg.widthProperty()).add(knot.radiusProperty()).add(H_GAP));
+        }else if (kti == 0) { // There is no argument left of the knot.
+            knot.setLayoutX(knot.getRadius());
+        }
+        
+        invalidateKnotPosition();
+    }
+    
+    /**
+     * Method to indicate that the position of the knot might have changed.
+     * 
+     * It repositions the arguments based on the current position of the knot.
+     */
+    public void invalidateKnotPosition() {
+        double bti = determineKnotIndex();
+        for (int i = 0; i < leftArguments.size(); i++) {
+            Region argument = leftArguments.get(i);
+            // Get only the percentage for this argument, clamped between 0 and 1, inverted.
+            double percentage = Math.min(Math.max(bti - i,  0), 1);            
+            percentage = Math.pow(percentage, 0.4); //Better looking curve
+            
+            //Updates the argument based on the percentage
+            argument.setTranslateX((H_GAP +knot.getRadius() * 2) * (1 - percentage));
+            argument.setOpacity(percentage);
+            argument.setVisible(percentage > 0);
+        }
+    }
+    
+    /**
+     * Method to indicate that the content in the argument Labels are possibly outdated.
+     */
+    public void invalidateArgumentContent() {
+        invalidateOutputContent();
+        invalidateInputContent();
+    }
+    
+    /**
+     * Method to indicate that the content in the output argument Label is possibly outdated.
+     */
+    public void invalidateOutputContent() {
+        String text = block.getOutputType().toHaskellType();
+        rightArgument.setText(text);        
+    }
+    
+    /**
+     * Method to indicate that the content in the input argument Labels are possibly outdated.
+     */
+    public void invalidateInputContent() {
+        for (InputArgument argument : leftArguments) {
+            argument.setInputText(argument.getInputAnchor().getType().toHaskellType());
+        }
+    }
+    
+    /**
+     * @return The InputArguments this ArgumentSpace has.
+     */
+    public List<InputArgument> getInputArguments() {
+        return leftArguments;
+    }
+    
+    /**
+     * @param index
+     * @return The input argument with the index as specified.
+     */
+    public InputArgument getInputArgument(int index) {
+        return leftArguments.get(index);
```

Comment:
Not really, this method returns an InputArgument, not an InputAnchor.

Did remove it though, axing could be Done by merging